tiledb/api/c_api/object/test/CMakeLists.txt

```diff
@@ -0,0 +1,59 @@
+#
+# tiledb/api/c_api/filesystem/test/CMakeLists.txt
+#
+# The MIT License
+#
+# Copyright (c) 2022 TileDB, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+find_package(Catch_EP REQUIRED)
+
+add_executable(unit_capi_object_type EXCLUDE_FROM_ALL)
+target_sources(unit_capi_object_type PUBLIC unit_capi_object_type.cc)
+target_include_directories(unit_capi_object_type PRIVATE ${TILEDB_EXPORT_HEADER_DIR})
+target_link_libraries(unit_capi_object_type
+  PUBLIC
+    TILEDB_CORE_OBJECTS
```

Comment:
I've been using the following criteria for object libraries and unit tests:
* Some object library outside of the API should precede an object library for an API object library.
* An object library that links separately should precede a separate unit test executable.
In other words, I've avoided creating unit tests for things that are not yet units. If you have to link the whole library to get a unit test executable, it's not a very good unit or perhaps not even a unit. In order to prevent tests that aren't unit tests from creeping into things that are called unit tests, I'd like to preserve the discipline that we don't link this way.

In this particular case the `object_*` parts of `StorageManager` could be easily separated out. It's worth a separate PR to do it, if needed. This PR is getting a bit crowded.

The type tests here are particularly egregious this way. They're really only testing symbol conversion and need very little code behind them.

Reply:
I removed all of the unnecessary linkage to get those object tests working since as you point out there's no real need to have all of that just for the string conversions. Once we finish up the StorageManager stuff moving things over with strict DAG dependency graphs and unit isolation will be easier.